pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

```diff
@@ -707,6 +709,19 @@ public void setObject(int parameterIndex, Object in, int targetSqlType, int scal
     }
   }
 
+  private <A> void setPrimitiveArray(int parameterIndex, A in) throws SQLException {
+    final PrimitiveArraySupport.ArrayToString<A> arrayToString = PrimitiveArraySupport.getArrayToString(in);
+
+    final int oid = arrayToString.getDefaultArrayTypeOid();
+
+    if (arrayToString.supportBinaryRepresentation() && connection.getPreferQueryMode() != PreferQueryMode.SIMPLE) {
+      bindBytes(parameterIndex, arrayToString.toBinaryRepresentation(in), oid);
```

Comment:
Agreed. Adding support for binary representation was a later thought and I did not revisit names.
I'll work on this a bit and see what I can come up with.